utils/sparse.py

```diff
@@ -200,20 +200,41 @@ def qat_active(self, epoch):
 
         return qat_start < epoch + 1
 
-    def reset_best(self, epoch):
+    def get_best_save_names(self, epoch):
+        # return a list of the names to save a "best" model under
+
+        save_names = ["best_overall"]
+
         if not self.enabled:
-            return False
+            return save_names
 
-        # if pruning is active or quantization just started, need to reset best checkpoint
-        # this is in case the pruned and/or quantized model do not fully recover
         pruning_start = math.floor(max([mod.start_epoch for mod in self.manager.pruning_modifiers])) \
-            if self.manager.pruning_modifiers else -1
+            if self.manager.pruning_modifiers else None
         pruning_end = math.ceil(max([mod.end_epoch for mod in self.manager.pruning_modifiers])) \
-            if self.manager.pruning_modifiers else -1
+            if self.manager.pruning_modifiers else None
         qat_start = math.floor(max([mod.start_epoch for mod in self.manager.quantization_modifiers])) \
```

Comment:
Shouldn't it be min here?

Reply:
yes, the qat_start should use min, pruning end should still use max